ihatemoney/currency_convertor.py

```diff
@@ -0,0 +1,41 @@
+import requests
+from diskcache import Cache
+
+
+class Singleton(type):
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
+        return cls._instances[cls]
+
+
+class CurrencyConverter(object, metaclass=Singleton):
+    # Convert all to USD
+    api_url = "https://api.exchangerate-api.com/v4/latest/USD"
+    _internal_cache = Cache()
+
+    def __init__(self, expire=86400):
+        self.expire = expire
+
+    def set_rate(self):
+        if "rates" not in self._internal_cache:
+            self._internal_cache.set(
+                "rates", requests.get(self.api_url).json()["rates"], expire=self.expire
+            )
+
+    def get_currencies(self):
+        self.set_rate()
+        return [(rate, rate) for rate in self._internal_cache.get("rates")]
+
+    def exchange_currency(self, amount, source_currency, dest_currency):
+        if source_currency == dest_currency:
+            return amount
+
+        self.set_rate()
+        source_rate = self._internal_cache.get("rates")[source_currency]
+        dest_rate = self._internal_cache.get("rates")[dest_currency]
+        new_amount = (amount / source_rate) * dest_rate
```

Comment:
Of course, since for now the amount is in float, we will output the result in float, but we need to start somewhere.

Reply:
While looking for python packages that might already be doing this, I stumbled upon https://github.com/limist/py-moneyed which seems to provide tooling for not doing weird operations between different currencies. Might be worth looking at?

Reply:
I am happy to add that, perhaps as a separate pull request. As this one is quite large, and I would also have to change the previous models. Which seems like a separate feature, where we upgrade all the amounts to decimal. As that is the only format the models would support natively.